lib/active_record_doctor/detectors/missing_foreign_keys.rb

```diff
@@ -23,30 +23,30 @@ def message(table:, column:)
       end
 
       def detect
-        each_table(except: config(:ignore_tables)) do |table|
-          each_column(table, except: config(:ignore_columns)) do |column|
-            # We need to skip polymorphic associations as they can reference
-            # multiple tables but a foreign key constraint can reference
-            # a single predefined table.
-            next unless looks_like_foreign_key?(column)
-            next if foreign_key?(table, column)
-            next if polymorphic_foreign_key?(table, column)
-
-            problem!(table: table, column: column.name)
+        table_models = models.select(&:table_exists?).group_by(&:table_name)
+        views = connection.views
+
+        table_models.each do |table, models|
+          next if ignored?(table, config(:ignore_tables))
+          next if views.include?(table)
+
+          concrete_models = models.reject do |model|
+            model.abstract_class?
           end
-        end
-      end
 
-      def foreign_key?(table, column)
-        connection.foreign_keys(table).any? do |foreign_key|
-          foreign_key.options[:column] == column.name
-        end
-      end
+          concrete_models.each do |model|
+            foreign_keys = connection.foreign_keys(table)
+            foreign_key_columns = foreign_keys.map { |key| key.options[:column] }
+            model.reflect_on_all_associations(:belongs_to).each do |association|
```

Comment:
There's `each_association` which we should be using here for the same reason.

Reply:
@gregnavis Thank you! Fixed the conflicts, updated based on your feedback, and pushed again!